File: interactivitycomjava/src/main/java/com/codekul/interactivitycomjava/SecondActivity.java
package com.codekul.interactivitycomjava;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.SeekBar;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {
    public static final String KEY_RES = "res";
    public static final String MSG = "";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        Bundle b = new Bundle();
        b = getIntent().getExtras();
        TextView textView = (TextView)findViewById(R.id.txt);
        textView.setTextSize(b.getFloat("defSize"));
        final SeekBar seekBar = (SeekBar)findViewById(R.id.skbrSize);
        seekBar.setProgress((int)b.getFloat("defSize"));
        seekBar.setOnSeekBarChangeListener(
                new SeekBar.OnSeekBarChangeListener() {
                    @Override
                    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                        TextView textView = (TextView)findViewById(R.id.txt);
                        textView.setTextSize((progress));
                        Log.i(MSG,""+seekBar.getProgress());


                    }

                    @Override
                    public void onStartTrackingTouch(SeekBar seekBar) {

                    }

                    @Override
                    public void onStopTrackingTouch(SeekBar seekBar) {

                    }
                }
        );


        Button button = (Button)findViewById(R.id.btnbck);
        button.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Log.i(MSG,"BUTTON METHOD");
                        Intent i =  new Intent();
                        Bundle bundle1 = new Bundle();
                        bundle1.putInt("ChkSize",seekBar.getProgress());
                        Log.i(MSG,""+seekBar.getProgress());
                        i.putExtras(bundle1);
                        setResult(Activity.RESULT_OK,i);
                        finish();
                    }
                }

        );



    }
}

File: interactivitycomkotlin/src/main/java/com/codekul/yog/interactivitycomkotlin/SecondActivity.kt
package com.codekul.yog.interactivitycomkotlin

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.SeekBar
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        setContentView(R.layout.activity_second)
        var bndl = intent.extras
        txt.textSize = bndl.getFloat("defSz")
        skbrSize.progress = bndl.getFloat("defSz").toInt()
        skbrSize.setOnSeekBarChangeListener(
                object : SeekBar.OnSeekBarChangeListener
                {
                    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                        txt.textSize=progress.toFloat()
                    }

                    override fun onStartTrackingTouch(seekBar: SeekBar?) {
                    }

                    override fun onStopTrackingTouch(seekBar: SeekBar?) {
                    }
                }
        )
        btnbck.setOnClickListener {
            val backInt = Intent()
            val bndl = Bundle()
            bndl.putInt("chSz",skbrSize.progress)
            backInt.putExtras(bndl)
            setResult(Activity.RESULT_OK,backInt)
            finish()
        }
    }
}
